keptn/keptn
# fix: Rename GitProxyInsecure to InsecureSkpTLS and pass it to upstream interactions

Part of #7363

Manually tested as follows:

1. Enable HTTPS on gitea, according to the following documentation: https://blog.differentpla.net/blog/2022/02/04/gitea-https/ (Note: a self signed certificate and a key can easily be generated on https://certificatetools.com)
2. Create project with https upstream URL pointing to gitea and `insecureSkipTLS` set to `false`: i.e. the credentials stored in the secret should look as follows:

```
{"user":"<user>","token":"<token>","remoteURI":"https://gitea-http:3000/<user>/<project>","insecureSkipTLS":false}
```

--> This will cause the resource service to fail when trying to communicate with the upstream (e.g. when cloning the repo)

3. Create project with https upstream and `insecureSkipTLS` set to true:

```
{"user":"<user>","token":"<token>","remoteURI":"https://gitea-http:3000/<user>/<project>","insecureSkipTLS":true}
```

--> With these settings, the resource service can communicate with the upstream over https even when gitea uses a self signed certificate

Note: This is a breaking change, since the `GitProxyInsecure` parameter has been renamed to `InsecureSkipTLS`, as this flag is not intended only for proxy communication, but for https upstream urls in general

Integration test run: https://github.com/keptn/keptn/runs/5972898882?check_suite_focus=true (the failing log forwarding test on one platform is not affected by the changes made in this PR and is probably a one-off, but we need to keep an eye on it in case it fails again)
